### PR TITLE
[Core][IO] added check for zero size in WriteIntegrationVectorContainerVariable …

### LINKED ISSUE #11885: [IO] In vtk_ouput.cpp non initialized variables are written to file, which crashes the computation
In vtk_ouput.cpp non initialized variables are written to file, which crashes the computation.

for example in vtk_ouput.cpp -> WriteIntegrationVectorContainerVariable, the following code is present which crashes

` // determining size of results
 const auto& r_process_info = mrModelPart.GetProcessInfo();
 std::vector<TVarType> tmp_result;
 rContainer.begin()->CalculateOnIntegrationPoints(rVariable, tmp_result, r_process_info);
 const int res_size = tmp_result[0].size();
`

if rContainer doesnt contain rVariable, tmp_result has size 0, thus tmp_result[0].size() is an invalid command which crashes the computation

Note that the computation does not crash when i request GIDOutput

### PR DESCRIPTION
**Pull request by aronnoordam**
added check to check if intergration point data has size==0 in vtk_output.cpp

### CODE REVIEW

**Review by loumalouomega — APPROVED** [review]

File: kratos/input_output/vtk_output.cpp [hunk 1]
```diff
@@ -964,6 +964,11 @@ void VtkOutput::WriteIntegrationVectorContainerVariable(
     const auto& r_process_info = mrModelPart.GetProcessInfo();
     std::vector<TVarType> tmp_result;
     rContainer.begin()->CalculateOnIntegrationPoints(rVariable, tmp_result, r_process_info);
+    
+    // if no results, return
+    if (tmp_result.size() == 0) {
+        return;
+    }
```
Comment (philbucher) [inline kratos/input_output/vtk_output.cpp:971]: in which case would one not have results? If an element does not write output for a variable?

I am mostly concerned how this works in MPI, when some partitions write this and some dont, do you know?

**Review by philbucher — CHANGES_REQUESTED** [review]
Blocking to prevent MPI problems

File: kratos/input_output/vtk_output.cpp [hunk 2]
```diff
@@ -964,6 +964,11 @@ void VtkOutput::WriteIntegrationVectorContainerVariable(
...
     const int res_size = tmp_result[0].size();
 
     rFileStream << rVariable.Name() << " " << res_size << " " << rContainer.size() << "  float\n";
```
Comment (philbucher) [inline kratos/input_output/vtk_output.cpp:974]: my concern is here, this would only be written on some ranks and thus make the files inconsistent

**Review by philbucher — APPROVED** [review]
I checked, I think it is ok, thanks for the fix

The handling for MPI with mixed entities is not ideal (already in master, not related to this MR), but should not really be a usecase

**Comment by aronnoordam** [comment]
okay, thanks for the review

**Pull request merged**